[FEATURE] add basic code quality checks via composer scripts and github actions

jpmschuler opened the pull request:
EXT:tea offers the results of the TYPO3 QA-Initiative for extensions, we should copy all that knowledge to have a defined code style which all IDEs pick up on, as well as have github actions auto-check code for incoming PRs.

Of course afterwards it will take some effort until everything is green for the first time, but having tools fix style is always a good choice.

This PR just adds the tools and CI/CD actions, it doesn't change the code itself.

opi99 commented on Classes/Utility/ApiHelperUtility.php:
```diff
@@ -53,7 +54,10 @@ public static function getBackendLayoutConfiguration($combinedBackendLayoutConfi
     public static function getConfiguration(string $combinedConfigurationIdentifier, string $handlerIdentifier): AbstractConfiguration
     {
-        list($placeIdentifier, $configurationIdentifier) = explode(':', $combinedConfigurationIdentifier);
+        if (strpos($combinedConfigurationIdentifier,':') === false) {
```
Missing space after comma.

jpmschuler replied:
done

opi99 reviewed (changes requested):
Do we need to mess inside the Resources/Private directory?

npm/yarn for what?

Does it need the /tools directory?

jpmschuler commented:
> Do we need to mess inside the Resources/Private directory?
No, fixed. Thought so too, changed EXT:tea: https://github.com/TYPO3-Documentation/tea/pull/445

> npm/yarn for what?
eslint and stylelint, the tools for fixing JS and CSS code style

> Does it need the /tools directory?
yes for the few tools that aren't manageable by composer and npm. There is a long discussion in the QA-team about that, but at least it is down to these few instead of everything. 

I'd say: stick to their long discussed standards instead of re-inventing the wheel.
